Source/WebCore/inspector/NetworkResourcesData.cpp

```diff
@@ -44,8 +45,24 @@ WTF_MAKE_TZONE_ALLOCATED_IMPL(NetworkResourcesData::ResourceData);
 
 using namespace Inspector;
 
-static const size_t maximumResourcesContentSize = 200 * 1000 * 1000; // 200MB
-static const size_t maximumSingleResourceContentSize = 50 * 1000 * 1000; // 50MB
+static const unsigned maximumResourcesContentSizeMB = 200; // 200MB
+static const unsigned maximumSingleResourceContentSizeMB = 50; // 50MB
+
+static size_t getMaximumResourcesContentSizeMB()
+{
+    static size_t maximumResourcesContentSize = maximumResourcesContentSizeMB;
+    static std::once_flag onceFlag;
+    std::call_once(onceFlag, [] {
+        const char* env = getenv("WEBKIT_INSPECTOR_MAXIMUM_RESOURCES_CONTENT_SIZE_MB");
+        if (env) {
+            auto value = parseInteger<unsigned>(StringView::fromLatin1(env));
+            if (value)
+                maximumResourcesContentSize = std::min(maximumResourcesContentSizeMB, std::max(maximumSingleResourceContentSizeMB, *value));
+        }
+    });
```

Comment:
i don't think this is desired for macOS/iOS/etc. so can we please at least wrap this with a `#if PLATFORM(...)`?

ideally we'd have it be a parameter when creating the `NetworkResourcesData` so that this can be plumbed out to a platform-specific `WebInspectorClient` implementation (see `InspectorPageAgent::overrideSetting` for an example of how something somewhat kinda like this has been done in the past)

Reply:
> i don't think this is desired for macOS/iOS/etc. so can we please at least wrap this with a `#if PLATFORM(...)`?
> 
> ideally we'd have it be a parameter when creating the `NetworkResourcesData` so that this can be plumbed out to a platform-specific `WebInspectorClient` implementation (see `InspectorPageAgent::overrideSetting` for an example of how something somewhat kinda like this has been done in the past)

@dcrousso : do you think would it be useful setting also in case of macOS/iOS/etc, I see it in embedded world but not sure about other platforms. In that case `#if PLATFORM(GTK) || PLATFORM(WPE)` is enough, right?

Reply:
I understand the embedded use case. I don't think we need this on iOS/macOS and related platforms.

Pulling runtime settings straight from env vars won't work in ports with a strict sandbox, nor is it easy to test the behavior that way. Therefore, I'd request you implement this as a configurable page override setting or platform-specific value.

Reply:
@burg : the latest implementation uses web preference `InspectorMaximumResourcesContentSize`